src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -165,7 +165,8 @@ protected override void OnKeyUp(KeyEventArgs e)
                             {
                                 // Search for the best version
                                 NuGetVersion rangeBestVersion = userRequestedVersionRange.FindBestMatch(versions);
-                                bool isBestOption = rangeBestVersion.ToString() == _versions.Items[_versions.SelectedIndex].ToString();
+                                DisplayVersion selectedVersion = _versions.Items.CurrentItem as DisplayVersion;
+                                bool isBestOption = rangeBestVersion.ToString() == selectedVersion.Version.ToString();
```

Comment:
`value as SomeType` will be null is `value` cannot be cast to `SomeType`. Since you then don't check `selectedVersion` to see if it's null, then this will throw a NullReferenceException, which is hard to figure out why a bug exists. But if you cast with `(DisplayVersion)value`, then we'll get an exception that clearly says something like "WhateverType can't be cast to SomeType`, making it much easier to understand.

Reply:
I created an issue for this https://github.com/NuGet/Home/issues/12738. First I was planning in adding that in this PR but I want to merge this in a QB insertion and adding that could be to much for the bar. Thanks for the reminder of adding that, I'll try to add that to every PR

Reply:
This pattern is followed in several places in this file. Would it be okay if Martin fixes all instances in a subsequent PR?

Reply:
Sure, a PR dedicated to fixing the pattern is ok, but I'd still like to see new PRs avoid the bad patterns in new or modified code. The so called "boy scout rule".

Reply:
Changed to use casting instead of as, there is a previous check for null, but the code is not clear. I really want to improve the code here in the next quality week. I did test the `#nullable` and there are a lot of improvements here for readability and nulls.